src/resmom/mom_server.c

```diff
@@ -1076,14 +1077,47 @@ is_request(int stream, int version)
 						LOG_NOTICE,
 						pjob->ji_qs.ji_jobid,
 						"Job discarded at request of Server");
+						if (pjob->ji_hook_running_bg_on) {
+							free(jobid);
+							jobid = NULL;
+							break;
+						}
 					(void)kill_job(pjob, SIGKILL);
+					phook_input = (mom_hook_input_t *)malloc(sizeof(mom_hook_input_t));
+					if (phook_input == NULL) {
+						log_err(errno, __func__, MALLOC_ERR_MSG);
+						goto err;
+					}
+					mom_hook_input_init(phook_input);
+					phook_input->pjob = pjob;
+					if ((phook_output = (mom_hook_output_t *)malloc(
+						sizeof(mom_hook_output_t))) == NULL) {
+							log_err(errno, __func__, MALLOC_ERR_MSG);
+							goto err;
+					}
+					mom_hook_output_init(phook_output);
 
-					mom_hook_input_init(&hook_input);
-					hook_input.pjob = pjob;
-					(void)mom_process_hooks(HOOK_EVENT_EXECJOB_END,
+					if ((phook_output->reject_errcode =
```

Comment:
Why does `mom_hook_output_t->reject_errcode` need to be a pointer? Can't it just be an integer?

Reply:
Existing declaration, is being used in other hook events too.